pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -449,11 +448,12 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 		case <-stillRunning.C:
 		}
 		ps.StillRunning(agentName, warningTime, errorTime)
-		if time.Since(wtTime) > 5*time.Minute { // wait for max of 5 minutes
-			waitEdgeNodeInfo = false
-		}
 	}
 	log.Noticef("processed GCComplete")
+	err = domainCtx.retrieveDeviceNodeName()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Can we log error and return, instead of fatal. Why crash device for this ?

Reply:
if we don't crash here, we can crash later. so better do it here.